File: 2_palindrome_number/palindrome-number.go
package main

import (
	"fmt"
)

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	given_int := x
	created_int := 0
	for x > 0 {
		reminder := x % 10
		created_int = (created_int * 10) + reminder
		x = x / 10
	}
	return created_int == given_int
}

func main() {
	fmt.Println(isPalindrome(10))
	fmt.Println(isPalindrome(101))
	fmt.Println(isPalindrome(11010))
}

File: 2_palindrome_number/palindrome-number.py
class Solution:
     def isPalindrome(self, x: int) -> bool:
          if x < 0:
            return False
          given_int = x
          created_int = 0
          while x > 0:
            reminder = x % 10
            print(reminder)
            created_int = (created_int * 10) + reminder
            x = int(x / 10)
          return created_int == given_int

if __name__ == "__main__":
    examples = [121, -121, 10]
    for index, example in enumerate(examples):
        print(f"Example {index}\nInput: {example} \nOutput: {Solution().isPalindrome(example)}")
